### ISSUE TITLE
More information about 'failed' and/or 'not started' consolidation process

### ISSUE THREAD

**Issue by AndrzejWisniewski** [aws/karpenter#3745]
### Tell us about your request

Would be great to have information (in the logs (DEBUG level?) or as events for the node) about checks/decisions which were made in consolidation's processes even if finally consolidation will be not done. Something like:
```
"Cannot deprovision node because of: pod-12342 (pdb), pod-12332 (topologySpreadConstraints -> topologyKey: topology.kubernetes.io/zone -> maxSkew: 1, ...) 
```

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

**Problem to solve:**

I can see in our clusters some nodes which are almost empty (% of CPU/MEM requested is near 5% or even less). But there is no any information in the events or logs about consolidation process (if such node was checked and karpenter decided it should be not consolidated? what is the reason why it's not consolidated? any issue with pdb (which pods)? any issues with topologySpreadConstraints (which pods, which constraint)? other?).

As I don't know full consolidation algorithm, without additional information, it's hard to investigate what is the reason that some nodes are not consolidated.

**What are you trying to do:**
I tried to figure out why some nodes are not consolidated and I think those should be.

### Are you currently working around this issue?

N/A

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by tzneal** [comment]
If you run `kubectl describe node node-name` , you should see events on the node that Karpenter creates indicating why it isn't consolidatable. E.g.

```
Events:
Type    Reason            Age                   From       Message
----    ------            ----                  ----       -------
Normal  Unconsolidatable  48m (x83 over 4d20h)  karpenter  can't remove without creating 2 nodes
```

**Comment by AndrzejWisniewski** [comment]
Thank you @tzneal for the quick reply 🙏 It's true that sometimes I can see such info (like you describe, or that it can not be consolidated because of pdb). 

But many times there is no information in events (and node is almost 'empty') - I believe this case can be because not all nodes get events about it - e.g. [like here](https://github.com/aws/karpenter-core/blob/main/pkg/controllers/deprovisioning/consolidation.go#L122). 
Or there is only information like 'not all pods would schedule"' but without any context (and when we have e.g. 20 pods there still run, and every of those pods has some specific pdb/topologySpreadConstraints/other it's really hard to find this one pod which is problematic :/).

Generally this gentle request is to ask for information about unconsolidated pods:
* always (not sometimes only for some specific cases)
* with more details (e.g. names of problematic pods and the reason why those pods are blocking consolidation)

**Comment by tzneal** [comment]
It should always display the events, there was a problem where it would not that should be fixed in v0.27.2.  If you still see this occurring, please re-open this issue. I had created https://github.com/aws/karpenter/issues/3746 for the feature request regarding more details on the event and am closing this in favor of that more specific feature request.

**Comment by AndrzejWisniewski** [comment]
Ok. Super thank you!
